src/client/eppo-client.ts

```diff
@@ -331,20 +350,54 @@ export default class EppoClient implements IEppoClient {
 
     // Compute variation for subject.
     const { subjectShards } = experimentConfig;
-    const { variations } = allocation;
+    const { variations, holdout } = allocation;
 
-    const shard = getShard(`assignment-${subjectKey}-${flagKey}`, subjectShards);
-    const assignedVariation = variations.find((variation) =>
-      isShardInRange(shard, variation.shardRange),
-    );
+    let assignedVariation;
+    let holdoutKey = null;
+    let holdoutVariation = null;
 
-    const internalAssignment = {
+    if (holdout && this.isInHoldoutSample(subjectKey, flagKey, experimentConfig, holdout)) {
+      const { statusQuo, shipped, keys } = holdout;
+      const holdoutShard = getShard(`holdout-assignment-${subjectKey}-${flagKey}`, subjectShards);
+      let matchingHoldout = keys.find((key) =>
+        isShardInRange(holdoutShard, key.statusQuoShardRange),
+      );
+      if (matchingHoldout) {
+        // The holdout serves status quo before rollout or within rollout split
+        holdoutKey = matchingHoldout?.holdoutKey;
+        assignedVariation = variations.find((variation) => variation.variationKey === statusQuo);
+        holdoutVariation = 'status_quo';
+      } else {
+        // If holdout is serving shipped, shipped and shippedShardRange are both defined
+        matchingHoldout = keys.find(
+          (key) =>
+            key.shippedShardRange != null && isShardInRange(holdoutShard, key.shippedShardRange),
```

Comment:
minor but its best practice to (almost) always use strict equality checking `!==`.

Although in this case--and seemingly contrary to the above best practice--since its an object I'd just check for truthiness like you did with holdout earlier